packages/procaptcha/src/modules/Manager.ts

```diff
@@ -197,46 +198,37 @@ export const Manager = (
-            console.log('providerApi', providerApi)
-            // get the captcha challenge and begin the challenge
-            const captchaApi = await loadCaptchaApi(contract, getRandomProviderResponse, providerApi)
-
-            console.log('captchaApi', captchaApi)
-            const challenge: GetCaptchaResponse = await captchaApi.getCaptchaChallenge()
-            console.log('challenge', challenge)
-            if (challenge.captchas.length <= 0) {
-                throw new Error('No captchas returned from provider')
-            }
-
-            // setup timeout
-            const timeMillis: number = challenge.captchas
-                .map((captcha) => captcha.captcha.timeLimitMs || 30 * 1000)
-                .reduce((a, b) => a + b)
-            const timeout = setTimeout(() => {
-                console.log('challenge expired after ' + timeMillis + 'ms')
-                events.onExpired()
-                // expired, disallow user's claim to be human
-                updateState({ isHuman: false, showModal: false, loading: false })
-            }, timeMillis)
-
-            // update state with new challenge
-            updateState({
-                index: 0,
-                solutions: challenge.captchas.map(() => []),
-                challenge,
-                showModal: true,
-                timeout,
-                blockNumber,
-            })
+            let challenge: GetCaptchaResponse
+            let blockNumber: u32
+
+            // Fetch the captcha challenge and retry up to 5 times if it fails, waiting for the next block before each retry
+            defer(() => fetchCaptchaChallenge(contract, account, config))
+                .pipe(retryWhen(retryStrategy({ maxRetryAttempts: 5, api: contract.api })))
+                .subscribe(
+                    (response) => {
+                        challenge = response.challenge
+                        blockNumber = response.blockNumber
+                        console.log('Challenge and block number fetched successfully', challenge, blockNumber)
+
+                        if (challenge.captchas.length <= 0) {
+                            throw new Error('No captchas returned from provider')
+                        }
+
+                        // setup timeout
+                        const timeout = timoutSetup(challenge, events, updateState)
+
+                        // update state with new challenge
+                        updateState({
+                            index: 0,
+                            solutions: challenge.captchas.map(() => []),
+                            challenge,
+                            showModal: true,
+                            timeout,
+                            blockNumber,
+                        })
+                    },
+                    (error) => console.error('Failed to fetch challenge and block number', error)
+                )
         } catch (err) {
```

Comment:
the stuff inside the defer call won't be caught by this catch now. Would it be better to put the whole of this fn inside the defer fn perhaps?